### PR TITLE
Only host with zfsvol can create SR of type zfsvol

### PR DESCRIPTION
**Pull request by gthvn1**

### CODE REVIEW

File: tests/storage/zfsvol/test_zfsvol_sr.py [hunk 1]
```diff
@@ -12,14 +12,17 @@
 
 pytestmark = pytest.mark.usefixtures("host_at_least_8_3")
 
+
+@pytest.mark.usefixtures("host_with_zfsvol")
```
Comment (stormi) [inline tests/storage/zfsvol/test_zfsvol_sr.py:16]: This is not needed at the moment since you also added the fixture to the only test in the class

Comment (gthvn1) [reply tests/storage/zfsvol/test_zfsvol_sr.py:16]: Ok. I thought that to use host_with_zfsvol as parameter I needed to also specify that it is a feature. I will remove it.

File: tests/storage/zfsvol/conftest.py [hunk 2]
```diff
@@ -8,6 +8,7 @@
 def host_with_zfsvol(host_with_saved_yum_state):
     host = host_with_saved_yum_state
     host.yum_install(['xcp-ng-xapi-storage-volume-zfsvol'])
+    host.restart_toolstack(verify=True)
```
Comment (gthvn1) [inline tests/storage/zfsvol/conftest.py:11]: we should also maybe add `systemctl daemon-reload` ?

Comment (Wescoeur) [reply tests/storage/zfsvol/conftest.py:11]: @stormi Can we directly add a `systemctl daemon-reload`  in the spec file of the smapi v3? Is there any contraindication to this?

**Comment by gthvn1** [comment]
As zfs-vol has bugs, running quicktest on zfs vol leaves TAP disk processes that we must clean up manually to be able to remove VDIs on SR storage and finally removes SR. So don't run quicktest until we fix the issue.

File: tests/storage/zfsvol/conftest.py (hunk 2 above)
Comment (stormi) [reply tests/storage/zfsvol/conftest.py:11]: Any RPM which updates a service file should usually call it, there are even macros that do it for you.

File: jobs.py [hunk 3]
```diff
@@ -341,6 +341,9 @@
 BROKEN_TESTS = [
     # not really broken but has complex prerequisites (3 NICs on 3 different networks)
     "tests/migration/test_host_evacuate.py::TestHostEvacuateWithNetwork",
+    # running quicktest on zfsvol generates dandling TAP devices that are hard to
```
Comment (stormi) [inline jobs.py:344]: I think you mean "dangling".
```suggestion
    # running quicktest on zfsvol generates dangling TAP devices that are hard to
```

File: jobs.py [hunk 4]
```diff
@@ -341,6 +341,9 @@
...
+    # cleanup. Bug need to be fixed before enabling quicktest on zfsvol.
```
Comment (stormi) [inline jobs.py:345]:
```suggestion
    # cleanup. Bug needs to be fixed before enabling quicktest on zfsvol.
```

File: tests/storage/zfsvol/test_zfsvol_sr.py [hunk 5]
```diff
@@ -19,7 +19,8 @@ class TestZfsvolSRCreateDestroy:
     and VM import.
     """
 
-    def test_create_and_destroy_sr(self, host, sr_disk_wiped):
+    def test_create_and_destroy_sr(self, host_with_zfsvol, sr_disk_wiped):
```
Comment (stormi) [inline tests/storage/zfsvol/test_zfsvol_sr.py:22]: AFAIK the order of fixtures determines which one is executed first. Here, we can ask ourselves: what do we want to fail first? The answer is: `sr_disk_wiped`. If the SR is missing, there's no point in starting to install RPM packages through the `host_with_zfsvol` fixture.

Thus I'd swap both parameters.

File: jobs.py (hunk 3 above)
Comment (gthvn1) [reply jobs.py:344]: dandling is more poetic but yes it is dangling like dangling pointers :)

File: tests/storage/zfsvol/test_zfsvol_sr.py (hunk 5 above)
Comment (gthvn1) [reply tests/storage/zfsvol/test_zfsvol_sr.py:22]: yes but where is this function called? Because all others tests are doing the same so I guess that there is some magic somewhere that calls it and we cannot do what we want?

Comment (stormi) [reply tests/storage/zfsvol/test_zfsvol_sr.py]: if other tests do the same, let's keep as is. We can improve them all together later if needed.

Comment (stormi) [reply tests/storage/zfsvol/test_zfsvol_sr.py]: > yes but where is this function called? 

pytest collects the fixtures, orders them, and then initializes them depending on their scope and what tests require them.

Comment (stormi) [reply tests/storage/zfsvol/test_zfsvol_sr.py]: What I observed additionnally is that when all other conditions have been treated by pytest and fixtures are at the same order and remain to be sorted, it looks like it then it oses the order in which it saw them. Could be an internal implementation detail and subject to change.

File: jobs.py (hunk 3 above)
Comment (stormi) [reply jobs.py]: It's also in the commit message

File: tests/storage/zfsvol/conftest.py (hunk 2 above)
Comment (stormi) [inline tests/storage/zfsvol/conftest.py:11]: This should be in a separate commit.

**Review by ydirson — APPROVED** [review]
lgtm

**Review by Wescoeur — APPROVED** [review]
